alien4cloud-core/src/main/java/alien4cloud/topology/TopologyService.java

```diff
@@ -424,13 +349,56 @@ public void isUniqueNodeTemplateName(Topology topology, String newNodeTemplateNa
         }
     }
 
+    public void updateDependencies(EditionContext context, CSARDependency newDependency) {
```

Comment:
The `updateDependencies()` method in ToscaContext does more than updating the dependencies : it actually cleans and resets the dependency map with the given values. Suggested action : rename the method to `setDependencies` to avoid confusion.